metabase/metabase
# Add `align-temporal-unit-with-param-type-and-value` multimethod

A follow-up on https://github.com/metabase/metabase/pull/34873. Required to resolve https://github.com/ClickHouse/metabase-clickhouse-driver/issues/216.

### Description

`align-temporal-unit-with-param-type` lacked access to `value`, which in certain cases made it impossible to make a decision on which temporal unit we should choose for the substitution. The newly introduced `align-temporal-unit-with-param-type-and-value` is intended to be a direct replacement.

### How to verify

The default behavior is unchanged and there are core tests to verify that. "Demo" tests were updated accordingly. As `align-temporal-unit-with-param-type` is marked as deprecated, the drivers implementing an override for it (probably ClickHouse only?) will get a warning, and there is a mention of the new method in the code, so it should be clear how to proceed.

### Checklist
